src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,122 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int)(DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int index = indexFor(hash(key), table.length);
+        for (Node<K,V> elem = table[index]; elem != null; elem = elem.next) {
+            Object k;
+            if (elem.hash == hash(key) && ((k = elem.key) == key || key.equals(k))) {
+                elem.value = value;
+                return;
+            }
+        }
+        addEntry(hash(key),key,value,index);
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            for (Node<K,V> elem = table[0]; elem != null; elem = elem.next) {
+                if (elem.key == null) {
+                    return elem.value;
+                }
+            }
+        }
+        int index = indexFor(hash(key), table.length);
+        for (Node<K,V> elem = table[index]; elem != null; elem = elem.next) {
+            Object currentKey;
+            if (elem.hash == hash(key) && ((currentKey = elem.key) == key
+                    || key.equals(currentKey))) {
+                return elem.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize(int newCapacity) {
+        Node[] newTable = new Node[newCapacity];
+        rehash(newTable);
+        table = newTable;
+        threshold = (int)(newCapacity * DEFAULT_LOAD_FACTOR);
+    }
+
+    private void rehash(Node[] newTable) {
```

Comment:
let's move this logic into resize that would be readable and we won't need to declare newCapacity twice

Reply:
done!